packages/ui/src/shared/ui-settings.ts

```diff
@@ -185,11 +206,15 @@ export interface UISettings {
     reUseEffectiveAclForPath?: string;
 }
 
+export type CustomVCSType = `custom_vsc_type_${string}`;
+
 export interface VCSSettings {
     id: string;
     name: string;
-    api: string;
-    type: 'github' | 'gitlab';
+    baseUrl: string;
+    auth: 'token' | 'none';
+    type: 'github' | 'gitlab' | CustomVCSType;
+    maxFileSize: number;
```

Comment:
It doesn't look like a good idea to suppose megabytes for a field with name `maxFileSize`. Bytes is more appropriate here.